File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *

setrecursionlimit(10**9)
input = stdin.readline


n = int(input())


times = [[0] * 2 for _ in range(n)]

for i in range(n):
    times[i][0] , times[i][1] = map(int,input().split())
    

times.sort(key = lambda t : (t[1] ,t[0])) # 끝나는 시간 기준

end_time = times[0][1]
cnt = 1


for i in range(1,n):
    s , e = times[i]
    
    if end_time  <= s : # 현재 회의 끝나는 시간이 <= 다음 회의 시작 시간 , 다음회의 가능 
        cnt += 1
        end_time = e


print(cnt)


    
    

File: 백준/Silver/1931. 회의실 배정/회의실 배정.swift
struct Time{
    
    let start:Int
    let end:Int
    
    
    public static func < (lhs:Self,rhs:Self) -> Bool {
        
        if lhs.end < rhs.end {
            return true
        }
        else if lhs.end == rhs.end {
            return lhs.start < rhs.start
        }
        
        return false
    }
    
}




let n = Int(readLine()!)!

var arr: [Time] = []

for _ in 0..<n {
    let input = readLine()!.split{$0 == " "}.map{Int($0)!}
    arr.append(Time(start: input[0],end: input[1]))
}

arr.sort(by: <)

var endTime: Int = 0
var ans: Int = 0

for time in arr {
    
    if endTime <= time.start {
        endTime = time.end
        ans += 1
    }
    
}

print(ans)
